code/modules/power/gravitygenerator.dm

```diff
@@ -311,13 +308,17 @@ var/const/GRAV_NEEDS_WRENCH = 3
 /obj/machinery/gravity_generator/main/Process()
 	if(stat & BROKEN)
 		return
+	if(charge_count >= 1)
+		var/pulse = 0.5 * sin(2 * M_PI * PULSE_FREQ * world.time) + 0.5
+		set_light(3+pulse, 3+pulse, "#8AD55D")
```

Comment:
I foresee unforeseen consequences.

Reply:
This proc is called every two ticks and it was working just like this for the entire time but it's work was dependened on a bug with infinite powering up that I fixed so I had to move it here
It's a great Idea with the best of intentions, what could possibly go wrong?